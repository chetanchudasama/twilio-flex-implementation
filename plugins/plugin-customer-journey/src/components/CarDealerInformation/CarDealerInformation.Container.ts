import { connect } from "react-redux";
import { Dispatch } from "redux";
import { getWorkerIsAvailableForOutboundCall } from "../../states/selectors";
import { AppState } from "../../states";
import { DispatchToProps, StateToProps } from "./CarDealerInformation.Props";
import CarDealerInformation from "./CarDealerInformation";

const mapStateToProps = (state: AppState): StateToProps => {
  return {
    canMakeCall: getWorkerIsAvailableForOutboundCall(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarDealerInformation);
