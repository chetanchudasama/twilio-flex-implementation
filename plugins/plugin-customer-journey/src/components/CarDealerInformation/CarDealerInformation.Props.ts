export interface StateToProps {
  canMakeCall: boolean;
}

export interface DispatchToProps {}
