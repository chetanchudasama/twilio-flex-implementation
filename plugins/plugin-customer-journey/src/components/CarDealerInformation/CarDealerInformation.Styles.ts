import { secondary, black } from "@common/components";
import styled from "react-emotion";

export const CarDealerInformationStyles = styled("div")({
  marginTop: 8,
  ".dealer-name": {
    fontSize: "16px",
    fontWeight: 600,
    paddingBottom: "8px",
    marginLeft: 5,
  },
  ".info-icon": {
    color: secondary,
    opacity: 0.4,
  },
  ".dealer-info": {
    fontSize: "14px",
    padding: "2px 10px 3px 10px",
    wordBreak: "break-word",
  },
  ".dealer-info-url": {
    fontSize: "14px",
    wordBreak: "break-word",
    padding: "2px 0px 3px 10px",
  },
  ".dealer-info-url a": {
    color: black,
  },
  ".dealer-phone, dealer-address": {
    display: "-webkit-box",
    WebkitBoxOrient: "vertical",
    WebkitLineClamp: 2,
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});
