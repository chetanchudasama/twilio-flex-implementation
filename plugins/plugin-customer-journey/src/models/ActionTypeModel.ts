export class ActionTypeModel {
  public id: number;

  public name: string;

  constructor() {
    this.id = -1;
    this.name = "";
  }
}
