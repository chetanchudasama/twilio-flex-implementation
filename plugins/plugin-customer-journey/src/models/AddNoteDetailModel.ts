export class AddNoteDetailModel {
  public applicationId: number;

  public content: string;

  public isImportant: boolean;

  public sendToBroker: boolean;

  constructor() {
    this.applicationId = -1;
    this.content = "";
    this.isImportant = false;
    this.sendToBroker = false;
  }
}
