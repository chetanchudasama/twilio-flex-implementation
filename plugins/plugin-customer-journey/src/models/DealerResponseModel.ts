import { DealerStateType } from "../common/enum";
import { CoordinatesResponseModel } from "./CoordinatesResponseModel";

export class DealerResponseModel {
  public dealerId: number;

  public dealerName: string;

  public building: string;

  public streetName: string;

  public town: string;

  public county: string;

  public postcode: string;

  public phone: string;

  public url: string;

  public rating: number | null;

  public coordinates: CoordinatesResponseModel;

  public state: DealerStateType;

  public declinedDate: Date | null;

  public declinedReason: string | null;

  constructor() {
    this.dealerId = 0;
    this.dealerName = "";
    this.building = "";
    this.streetName = "";
    this.town = "";
    this.county = "";
    this.postcode = "";
    this.phone = "";
    this.url = "";
    this.rating = null;
    this.coordinates = new CoordinatesResponseModel();
    this.state = 3;
    this.declinedDate = null;
    this.declinedReason = "";
  }
}
