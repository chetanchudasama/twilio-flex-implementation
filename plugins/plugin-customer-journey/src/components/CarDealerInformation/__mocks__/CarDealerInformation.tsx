import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
