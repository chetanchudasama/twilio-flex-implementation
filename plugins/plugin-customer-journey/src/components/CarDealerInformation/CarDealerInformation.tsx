import { Grid } from "@material-ui/core";
import LanguageIcon from "@material-ui/icons/Language";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import StarIcon from "@material-ui/icons/Star";
import { Actions, IconButton } from "@twilio/flex-ui";
import React from "react";
import { DealerResponseModel } from "../../models/DealerResponseModel";
import { StateToProps, DispatchToProps } from "./CarDealerInformation.Props";
import { CarDealerInformationStyles } from "./CarDealerInformation.Styles";

export interface CarDealerInformationProp {
  dealerInfo: DealerResponseModel;
  distance?: number;
}

export type Props = StateToProps & DispatchToProps & CarDealerInformationProp;

const CarDealerInformation: React.FC<Props> = ({
  dealerInfo,
  distance,
  canMakeCall,
}) => {
  const triggerOutboundCall = () => {
    if (canMakeCall && dealerInfo.phone) {
      const payload = {
        destination: dealerInfo.phone,
      };
      Actions.invokeAction("StartOutboundCall", payload);
    }
  };

  return (
    <CarDealerInformationStyles>
      <Grid container>
        <Grid item xs={12} className="dealer-name">
          {dealerInfo.dealerName}
        </Grid>
        <Grid item container>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <LocationOnIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-address">
                {dealerInfo.town + (distance ? ` (${distance} miles)` : "")}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <StarIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info dealer-rating">
                {dealerInfo.rating ?? " -"}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <IconButton
                  icon="Call"
                  className="info-icon"
                  disabled={!canMakeCall}
                  onClick={() => triggerOutboundCall()}
                />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-phone">
                {dealerInfo.phone}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <LanguageIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info-url">
                <a href={dealerInfo.url} target="blank">
                  Website
                </a>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </CarDealerInformationStyles>
  );
};

export default CarDealerInformation;
