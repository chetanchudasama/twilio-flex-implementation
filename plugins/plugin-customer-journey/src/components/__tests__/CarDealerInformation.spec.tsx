import React from "react";
import { create } from "react-test-renderer";
import CarDealerInformation, {
  Props,
} from "../CarDealerInformation/CarDealerInformation";
import { DealerResponseModel } from "../../models/DealerResponseModel";

jest.mock("@twilio/flex-ui", () => {
  return {
    IconButton: jest.fn(({ children }) => <div>{children}</div>),
  };
});

const dealer: DealerResponseModel = Object.assign(new DealerResponseModel(), {
  dealerId: 1,
  dealerName: "MR john",
  county: "UK",
  building: "13",
  streetName: "46 Grenoble Road",
  town: "BREADSTONE",
  rating: 4,
  phone: "+447700870088",
  url: "https://namesbee.com/car-dealership-names/",
  postcode: "GL13 1XD",
});

const props: Props = {
  dealerInfo: dealer,
  distance: undefined,
  canMakeCall: false,
};

describe("CarDealerInformation", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<CarDealerInformation {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render props value correctly in UI", () => {
    const wrapper = create(<CarDealerInformation {...props} />).root;

    expect(
      wrapper.findByProps({
        className: "dealer-name",
      }).props.children
    ).toContain(props.dealerInfo.dealerName);

    expect(
      wrapper.findByProps({
        className: "dealer-info dealer-address",
      }).props.children
    ).toContain(props.dealerInfo.town);

    expect(
      wrapper.findByProps({
        className: "dealer-info dealer-phone",
      }).props.children
    ).toContain(props.dealerInfo.phone);

    expect(
      wrapper.findByProps({
        className: "dealer-info dealer-rating",
      }).props.children
    ).toEqual(props.dealerInfo.rating);

    expect(wrapper.findByType("a").props.href).toContain(props.dealerInfo.url);
  });

  it("when value of distance prop is null, should render only town name", () => {
    const wrapper = create(<CarDealerInformation {...props} />).root;

    expect(
      wrapper.findByProps({
        className: "dealer-info dealer-address",
      }).props.children
    ).toContain(props.dealerInfo.town);
  });

  it("when value of distance prop is not null, should render town name with distance", () => {
    props.distance = 2.6;
    const wrapper = create(<CarDealerInformation {...props} />).root;

    expect(
      wrapper.findByProps({
        className: "dealer-info dealer-address",
      }).props.children
    ).toContain(`${props.dealerInfo.town} (${props.distance} miles)`);
  });
});
