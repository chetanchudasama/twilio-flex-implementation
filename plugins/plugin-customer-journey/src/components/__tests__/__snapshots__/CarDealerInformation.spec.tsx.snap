// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDealerInformation renders correctly on load 1`] = `
<div
  className="css-qzbvbw"
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 dealer-name"
    >
      MR john
    </div>
    <div
      className="MuiGrid-container-1 MuiGrid-item-2"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
      >
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-1-30"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-98 MuiSvgIcon-fontSizeSmall-105 info-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-11-40 dealer-info dealer-address"
          >
            BREADSTONE
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-98 MuiSvgIcon-fontSizeSmall-105 info-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 dealer-info dealer-rating"
          >
            4
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
      >
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-1-30"
          >
            <div />
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-11-40 dealer-info dealer-phone"
          >
            +447700870088
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-98 MuiSvgIcon-fontSizeSmall-105 info-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 dealer-info-url"
          >
            <a
              href="https://namesbee.com/car-dealership-names/"
              target="blank"
            >
              Website
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
